.github/workflows/ci.yaml

```diff
@@ -34,7 +34,7 @@ env:
   HA_SHORT_VERSION: 2023.3
   DEFAULT_PYTHON: "3.10"
   ALL_PYTHON_VERSIONS: "['3.10']"
-  MARIADB_VERSIONS: "['mariadb:10.9.3']"
+  MARIADB_VERSIONS: "['mariadb:10.6.10','mariadb:10.9.3']"
```

Comment:
We only have to test the minimal supported, right?

Reply:
I think just 10.6.9 or 10.6.10 is good enough as it's the lowest "good" version

Reply:
But will it cover all the various branching?

Reply:
We would need to test with 10.3 and 10.6.x to cover all the branches

Reply:
I've adjusted to use `10.3.32` and `10.6.10`
However, the tests fail on `10.3.32`: 
```console
=========================== short test summary info ============================
FAILED tests/components/recorder/test_purge.py::test_purge_many_old_events - assert 6 == 5
 +  where 6 = <bound method Query.count of <sqlalchemy.orm.query.Query object at 0x7f65e9fdecb0>>()
 +    where <bound method Query.count of <sqlalchemy.orm.query.Query object at 0x7f65e9fdecb0>> = <sqlalchemy.orm.query.Query object at 0x7f65e9fdecb0>.count
```

Reply:
I really think we should test with a new version too, not only with old versions.
10.3 is from 2017, 10.6 is from 2021

Reply:
We also need to merge in dev to pickup https://github.com/home-assistant/core/pull/87753

Reply:
> I've adjusted to use `10.3.32` and `10.6.10` However, the tests fail on `10.3.32`:
> 
> ```
> =========================== short test summary info ============================
> FAILED tests/components/recorder/test_purge.py::test_purge_many_old_events - assert 6 == 5
>  +  where 6 = <bound method Query.count of <sqlalchemy.orm.query.Query object at 0x7f65e9fdecb0>>()
>  +    where <bound method Query.count of <sqlalchemy.orm.query.Query object at 0x7f65e9fdecb0>> = <sqlalchemy.orm.query.Query object at 0x7f65e9fdecb0>.count
> ```

Might have to install 10.3.x locally to figure out what’s going on here

Reply:
Still fails on 10.3.32